podpac/datalib/soilscape.py

```diff
@@ -0,0 +1,663 @@
+import logging
+import datetime
+
+import traitlets as tl
+import numpy as np
+
+import podpac
+
+_logger = logging.getLogger(__name__)
+
+SOILSCAPE_FILESERVER_BASE = "https://thredds.daac.ornl.gov/thredds/fileServer/ornldaac/1339"
+CRS = "+proj=merc +vunits=cm"  # TODO
```

Comment:
@mpu-creare Can you figure out what the CRS is?

Reply:
Do you know how to use WGS84 with vunits to proj4?

Reply:
Using the WGS84 datum with vunits *does* add the h[up] axis with the correct units, but it doesn't register as `is_vertical` by PROJ4. The PROJ4 representation drops the vunits, but e.g. the WKT representation does keep them.

```
>>> p = pyproj.Proj('+proj=longlat +datum=WGS84 +vunits=cm')
>>> p
Proj('+proj=longlat +datum=WGS84 +no_defs', preserve_units=True)
>>> p.crs
<Geographic 3D CRS: +proj=longlat +datum=WGS84 +vunits=cm +type=crs>
Name: unknown
Axis Info [ellipsoidal]:
- lon[east]: Longitude (degree)
- lat[north]: Latitude (degree)
- h[up]: Ellipsoidal height (centimetre)
Area of Use:
- undefined
Datum: World Geodetic System 1984
- Ellipsoid: WGS 84
- Prime Meridian: Greenwich
>>> p.crs.is_vertical
False
```

Reply:
```
>>> pyproj.Proj('+proj=longlat +datum=WGS84 +vunits=cm').crs.axis_info                                                           
[Axis(name=Longitude, abbrev=lon, direction=east, unit_auth_code=EPSG, unit_code=9122, unit_name=degree),
 Axis(name=Latitude, abbrev=lat, direction=north, unit_auth_code=EPSG, unit_code=9122, unit_name=degree),
 Axis(name=Ellipsoidal height, abbrev=h, direction=up, unit_auth_code=EPSG, unit_code=1033, unit_name=centimetre)]
```

Reply:
Okay, it looks like what I posted above works (including transformations), I just need to change the `is_vertical` checks to something else like `axis_info`, because `is_vertical` is not reliable.

Reply:
Ugh.

So, there is a difference between "Ellipsoidal height" and "Gravity-related height", and when you specify `+vunits` with WGS84, it chooses ellipsoidal height for you, but other times, it selects gravity-related.

```
>>> pyproj.Proj('+proj=merc +vunits=cm').crs.axis_info                                                                           
[Axis(name=Easting, abbrev=E, direction=east, unit_auth_code=EPSG, unit_code=9001, unit_name=metre),
 Axis(name=Northing, abbrev=N, direction=north, unit_auth_code=EPSG, unit_code=9001, unit_name=metre),
 Axis(name=Gravity-related height, abbrev=H, direction=up, unit_auth_code=EPSG, unit_code=1033, unit_name=centimetre)]
>>> pyproj.Proj('+proj=longlat +datum=WGS84 +vunits=cm').crs.axis_info                                                           
[Axis(name=Longitude, abbrev=lon, direction=east, unit_auth_code=EPSG, unit_code=9122, unit_name=degree),
 Axis(name=Latitude, abbrev=lat, direction=north, unit_auth_code=EPSG, unit_code=9122, unit_name=degree),
 Axis(name=Ellipsoidal height, abbrev=h, direction=up, unit_auth_code=EPSG, unit_code=1033, unit_name=centimetre)]
```

Is "Gravity-related height" orthometric/geoid height? Will there be slight differences between the two? I tried transforming between them to find out, and instead, looks like a bug:

```
>>> a = pyproj.Proj('+proj=merc +vunits=cm')                                                                                      
>>> b = pyproj.Proj('+proj=longlat +datum=WGS84 +vunits=cm')                                                                      
>>> transformer = pyproj.Transformer.from_proj(a, b, always_xy=True)  
>>> transformer.transform([20, 20, 20], [100, 100, 200], [1, 2, 1])                                                               
([0.00017966305682390427, 0.00017966305682390427, 0.00017966305682390427],
 [0.0009043694751898239, 0.0009043694751898239, 0.0018087389501633701],
 [100.0, 200.0, 100.0])
```

The other direction, the vertical coordinates are preserved (no bug, no difference between gravity-related and ellipsoidal, at least at the lat_lon I tried).

Not sure what to do here... I can just use `'+proj=longlat +datum=WGS84 +vunits=cm'`, update the `Coordinates` to check for `any(axis.direction == 'up' for axis in crs.axis_info)` instead of `is_vertical`, file a bug report with PROJ4, and say good enough. *(estimate: 1 hour)*

Another alternative is to punt: report the soilscape data in 3 outputs instead of a coordinate dimension so that the crs doesn't have to deal with it. *(estimate: 2 hours)*

Last option is to revert to an `alt_units` attribute, and detect alt_units from crs when possible but store the alt_units separately so that we can be flexible. *(estimate: 4 hours)*

@mpu-creare preference?

Reply:
I vote for: 
> I can just use '+proj=longlat +datum=WGS84 +vunits=cm', update the Coordinates to check for any(axis.direction == 'up' for axis in crs.axis_info) instead of is_vertical, file a bug report with PROJ4, and say good enough.

Reply:
FWIW, here is the bug report that I submitted: https://github.com/pyproj4/pyproj/issues/688

Reply:
Pushed upstream to https://github.com/OSGeo/PROJ/issues/2319